packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx

```diff
@@ -204,6 +221,80 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
+    let cornerIndexes = [];
+
+    if (weekType === 'firstWeek') {
+
+      if (week[DAYS_IN_WEEK - 2].isInMonth) {
+        let firstDayOfMonth = addDays(navigatedDate, - navigatedDate.getDate() + 1);
+        // Account for firstDayOfWeek not beeing Sunday
+        let firstDayIndex = firstDayOfMonth.getDay() - firstDayOfWeek;
+        firstDayIndex = firstDayIndex < 0 ? firstDayIndex + DAYS_IN_WEEK : firstDayIndex;
+        cornerIndexes.push(firstDayIndex);
+      }
+      cornerIndexes.push(DAYS_IN_WEEK - 1);
+    } else {
+      if (week[1].isInMonth) {
+        let daysInMonth = new Date(navigatedDate.getFullYear(), navigatedDate.getMonth() + 1, 0).getDate();
+        let lastDayofMonth = addDays(navigatedDate, daysInMonth - navigatedDate.getDate());
+        // Account for firstDayOfWeek not beeing Sunday
+        let lastDayIndex = lastDayofMonth.getDay() - firstDayOfWeek;
+        lastDayIndex = lastDayIndex < 0 ? lastDayIndex + DAYS_IN_WEEK : lastDayIndex;
+        cornerIndexes.push(lastDayIndex);
+      }
+
+      cornerIndexes.push(0);
+    }
+
+    return cornerIndexes;
+  }
+
+  private _getWeekCornerStyles(weeks: IDayInfo[][]) {
+    let { firstDayOfWeek, navigatedDate, dateRangeType } = this.props;
+
+    let weekCornersStyled: any = {};
+    let numberOfWeeks = weeks.length;
+    let firstWeek = weeks[0];
+    let lastWeek = weeks[numberOfWeeks - 1];
+    let indexesFirstWeek = this._findCornerIndexes(firstWeek, 'firstWeek');
+    let indexesLastWeek = this._findCornerIndexes(lastWeek, 'lastWeek');
+
+    if (indexesFirstWeek.length === 1) {
+      weekCornersStyled[0 + '_' + indexesFirstWeek[0]] = 'ms-DatePicker-singleTopDate ' + styles.singleTopDate;
+    } else {
+      weekCornersStyled[0 + '_' + indexesFirstWeek[0]] = 'ms-DatePicker-topLeftCornerDate ' + styles.topLeftCornerDate;
+      weekCornersStyled[0 + '_' + indexesFirstWeek[1]] = 'ms-DatePicker-topRightCornerDate ' + styles.topRightCornerDate;
+    }
+
+    // check if second week needs corner styles
+    if (!weeks[0][0].isInMonth) {
+      weekCornersStyled[1 + '_' + 0] = 'ms-DatePicker-topLeftCornerDate ' + styles.topLeftCornerDate;
+    }
+
+    let lastWeekIndex = numberOfWeeks - 1;
+    if (indexesLastWeek.length === 1) {
+      weekCornersStyled[lastWeekIndex + '_' + indexesLastWeek[0]] = 'ms-DatePicker-singleBottomDate ' + styles.singleBottomDate;
+    } else {
+      weekCornersStyled[lastWeekIndex + '_' + indexesLastWeek[0]] = 'ms-DatePicker-bottomRightCornerDate ' + styles.bottomRightCornerDate;
+      weekCornersStyled[lastWeekIndex + '_' + indexesLastWeek[1]] = 'ms-DatePicker-bottomLeftCornerDate ' + styles.bottomLeftCornerDate;
+    }
+
+    // check if second-to-last week needs corner styles
+    if (!weeks[lastWeekIndex][DAYS_IN_WEEK - 1].isInMonth) {
+      weekCornersStyled[(lastWeekIndex - 1) + '_' + (DAYS_IN_WEEK - 1)] = 'ms-DatePicker-bottomRightCornerDate ' + styles.bottomRightCornerDate;
+    }
+
+    return weekCornersStyled;
```

Comment:
There are a bunch of undefined values in the output of this function, it looks like none of the styles referenced are defined...

Reply:
I also sent you a slightly updated version of this code that is a bit cleaner